pkg/auth/providerrefresh/refresher_test.go

```diff
@@ -0,0 +1,192 @@
+package providerrefresh
+
+import (
+	"context"
+	"testing"
+
+	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/providers"
+	"github.com/rancher/rancher/pkg/auth/providers/common"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	managementv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+)
+
+func Test_refreshAttributes(t *testing.T) {
+	tests := []struct {
+		name    string
+		user    *v3.User
+		attribs *v3.UserAttribute
+		tokens  []*v3.Token
+		want    *v3.UserAttribute
+	}{
+		{
```

Comment:
Nit: 

I think right now we have test cases for:
- No token
- 1 login no derived
- 1 derived no login

It might be good to include a case with 1 login and 1 derived (and assert that it uses what is on the login). Especially since I would expect this to be a relatively common case.

Reply:
Added this test